README
MySQL Server SNMP Agent.
Copyright (c) 2008,2009 Brice Figureau <brice.figureau@daysofwonder.com>

PREAMBLE
========

mysql-snmp is a MySQL Net-SNMP agent written in Perl, and using the Net-Snmp 
Perl bindings.

It connects to a mysql server and returns information to Net-SNMP when needed.
It parses SHOW INNODB STATUS and other MySQL status variables.

This work is based on Baron Schwartz Mysql Cacti Templates, available at:
http://code.google.com/p/mysql-cacti-templates/

This work is based on the blog post about a SNMP agent:
http://mysqldump.azundris.com/archives/63-Sysadmins-Nightly-Mental-Pain-SNMP.html

This software is released under the GPL v2.


REQUIREMENTS
============

This package requires Net-SNMP version 5.4.3 or better. Previous versions have bugs preventing
returning 64 bits MIBs objects correctly from a perl agentX.

INSTALL
=======

Debian
------

Please build the package:
$ dpkg-buildpackage -rfakeroot -uc- b

and install it
Then, modify /etc/default/mysql-snmp

Other
-----

Modify the Makefile, and:
$ sudo make install

Then create an init.d startup script. You can look to the debian one
in debian/init.d

Install the MIB file in your mibs directory.

CONFIGURE
=========

Configure net-snmp
------------------

Add:
master agentx
to /etc/snmp/snmpd.conf, and restart snmpd
Make sure the mysql-snmp agent is able to open the agentX socket (permissions
and location of said socket can be managed with the snmpd.conf agentXPerms
directive).

Configure mysql-snmp
--------------------

Under debian, the daemon can be configured with /etc/default/mysql-snmp.
On all platform, the configuration is done with command line arguments:

-h|--host HOST          connect to mysql HOST
-P|--port PORT          port to connect to (default 3306)
-u|--user DBUSER        use DBUSER as user to connect to mysql
-p|--passwordi DBPASS   use DBPASS as password to connect to mysql
-c|--config FILE        read MySQL configuration from FILE

-m|--master             check master
-s|--slave              check slave
-b|--heartbeat DB.TABLE table for maatkit mk-heartbeat 
--oid OID               registering OID
-l|--process-list|--procs
                        enable the MySQL process list
-i|--refresh            refresh interval in seconds

You can specify the mysql connection parameters in a config file using my.cnf format, like this:
For example:

[client]
host=dbserver
port=3306
user=monitor
password=secret


OPENNMS
=======

You'll find in the opennms directory the various configuration snippets
that can be used to let opennms graph the value reported by this agent.

datacollection-config.xml
-------------------------

Add the content of opennms/mysql.datacollection-config.xml to
your datacollection-config.xml file.

Then add this group to your hosts:
<systemDef name="Net-Snmp">
  <sysoidMask>.1.3.6.1.4.1.8072.3.</sysoidMask>
  <collect>
    <includeGroup>mib2-host-resources-system</includeGroup>
    <includeGroup>mib2-host-resources-memory</includeGroup>
    <includeGroup>net-snmp-disk</includeGroup>
    <includeGroup>ucd-loadavg</includeGroup>
    <includeGroup>ucd-memory</includeGroup>
    <includeGroup>ucd-sysstat</includeGroup>
    <includeGroup>mysql</includeGroup>
  </collect>
</systemDef>

And now, add the graph definitions from the opennms/mysql.snmp-graph.properties file 
to your snmp-graph.properties.


debian/changelog
mysql-snmp (1.2-1) unstable; urgency=low

  * new version
  * some new graphs (innodb row lock times, innodb sem times, key buffer 
  usage)
  * renamed all mysql-agent to mysql-snmp (thanks to Robin Bowes)
  * activate process list fetching with --procs (thanks to Robin Bowes)

 -- brice <brice+debian@daysofwonder.com>  Mon, 15 Aug 2011 17:55:37 +0200

mysql-snmp (1.0-1) unstable; urgency=low

  * new upstream version

 -- brice <brice+debian@daysofwonder.com>  Sun, 10 Jan 2010 17:23:15 +0100

mysql-snmp (1.0rc1-1) unstable; urgency=low

  * new upstream version (more innodb graphs)

 -- brice <brice+debian@daysofwonder.com>  Sat, 31 Oct 2009 13:49:15 +0100

mysql-snmp (0.8-1) unstable; urgency=low

  * new upstream version

 -- brice <brice+debian@daysofwonder.com>  Sat, 24 Oct 2009 18:55:15 +0200

mysql-snmp (0.7-1) unstable; urgency=low

  * new upstream version
  * moved manpage to top directory

 -- brice <brice+debian@daysofwonder.com>  Sat, 24 Oct 2009 17:12:44 +0200

mysql-snmp (0.6-1) unstable; urgency=low

  * returns No Such OID in case of refresh issues (should prevent rrd spikes)

 -- brice <brice+debian@daysofwonder.com>  Sat,  2 Aug 2008 15:58:44 +0200

mysql-snmp (0.5-1) unstable; urgency=low

  * fixed myConnections type 

 -- brice <brice+debian@daysofwonder.com>  Tue, 22 Jul 2008 19:28:34 +0200

mysql-snmp (0.4-1) unstable; urgency=low

  * installs mysql-snmp.default 

 -- brice <brice+debian@daysofwonder.com>  Sat, 19 Jul 2008 14:12:54 +0200

mysql-snmp (0.4) unstable; urgency=high

  * new version compatible with net-snmp 5.4.1 

 -- brice <brice+debian@daysofwonder.com>  Sat, 19 Jul 2008 14:02:53 +0200

mysql-snmp (0.3) unstable; urgency=low

  * init.d bugfixes

 -- brice <brice+debian@daysofwonder.com>  Fri, 18 Jul 2008 16:13:32 +0200

mysql-snmp (0.2) unstable; urgency=low

  * new release with better code

 -- brice <brice+debian@daysofwonder.com>  Fri, 18 Jul 2008 15:49:53 +0200

mysql-snmp (0.1-1) unstable; urgency=low

  * Initial release

 -- brice <brice+debian@daysofwonder.com>  Thu, 17 Jul 2008 15:45:19 +0200



debian/init.d
#! /bin/sh
#
# Copyright (c) 2008, 2009 Brice Figureau
#
# This init script is based on:
# the skeleton	example file to build /etc/init.d/ scripts.
#		This file should be used to construct scripts for /etc/init.d.
#
#		Written by Miquel van Smoorenburg <miquels@cistron.nl>.
#		Modified for Debian 
#		by Ian Murdock <imurdock@gnu.ai.mit.edu>.
#
# Version:	@(#)skeleton  1.9  26-Feb-2001  miquels@cistron.nl
#
### BEGIN INIT INFO
# Provides:          mysql-snmp
# Required-Start:    $syslog
# Required-Stop:     $syslog
# Should-Start:      $local_fs $remote_fs $network $named $time
# Should-Stop:       $local_fs $remote_fs $network $named $time
# Default-Start:     2 3 4 5
# Default-Stop:      0 1 6
# Short-Description: Start and stop the mysql database server snmp agent
# Description:       Controls the launch/stop of Mysql SNMP agent.
### END INIT INFO

PATH=/sbin:/bin:/usr/sbin:/usr/bin
DAEMON=/usr/sbin/mysql-smnp
DAEMON2="/usr/bin/perl"
NAME=mysql-snmp
DESC="Mysql SNMP Agent"
PID_FILE=/var/run/mysql-snmp.pid

test -x $DAEMON || exit 0

. /lib/lsb/init-functions

# Include mysql-snmp defaults if available
if [ -f /etc/default/mysql-snmp ] ; then
	. /etc/default/mysql-snmp
fi

set -e

START="--start --quiet --pidfile ${PID_FILE} --name ${NAME} --startas ${DAEMON}"
STOP="--stop --quiet --pidfile ${PID_FILE} --name ${NAME}"

case "$1" in
  start)
		echo -n "Starting $DESC: "
    if start-stop-daemon ${START} -- --daemon-pid ${PID_FILE} ${DAEMON_OPTS} >/dev/null ; then
            echo "${NAME}."
    else
            if start-stop-daemon --test ${DAEMON_OPTS} >/dev/null 2>&1; then
                    echo "(failed)."
                    exit 1
            else
                    echo "(already running)."
                    exit 0
            fi
    fi
    ;;
  stop)
		log_daemon_msg "Stopping $DESC" "$NAME"
		killproc -p ${PID_FILE} $DAEMON2
		log_end_msg $?
    ;;
  restart)
    $0 stop
    exec $0 start
    ;;
  force-reload)
    $0 restart
    ;;
  status)
    pidofproc -p ${PID_FILE} $DAEMON2 >/dev/null
    status=$?
    if [ $status -eq 0 ]; then
            log_success_msg "mysql-snmp server is running."
    else
            log_failure_msg "mysql-snmp server is not running."
    fi
    exit $status
    ;;
  *)
    echo "Usage: $0 start|stop|restart|status|force-reload"
    exit 1
    ;;
esac

exit 0


opennms/tools/cacti2MIB.pl
# mysql-snmp: cacti2MIB.pl
#
# This file contains the mapping from MySQL cacti templates to the MIB names, value and
# and types. This file is used to generate the opennms configuration.
#
# (c) Copryright 2009 - Brice Figureau
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program. If not, see <http://www.gnu.org/licenses/>.

{   name        => 'Cacti 2 MIB',
    short_names => {
        'Key_read_requests'         => {mib => 'myKeyReadRequests',             order => 1,   type => 'Counter32'},
        'Key_reads'                 => {mib => 'myKeyReads',                    order => 2,   type => 'Counter32'},
        'Key_write_requests'        => {mib => 'myKeyWriteRequests',            order => 3,   type => 'Counter32'},
        'Key_writes'                => {mib => 'myKeyWrites',                   order => 4,   type => 'Counter32'},
        'history_list'              => {mib => 'myHistoryList',                 order => 5,   type => 'Gauge32'},
        'innodb_transactions'       => {mib => 'myInnodbTransactions',          order => 6,   type => 'Counter64'},
        'read_views'                => {mib => 'myReadViews',                   order => 7,   type => 'Gauge32'},
        'current_transactions'      => {mib => 'myCurrentTransactions',         order => 8,   type => 'Gauge32'},
        'locked_transactions'       => {mib => 'myLockedTransactions',          order => 9,   type => 'Gauge32'},
        'active_transactions'       => {mib => 'myActiveTransactions',          order => 10,  type => 'Gauge32'},
        'pool_size'                 => {mib => 'myPoolSize',                    order => 11,  type => 'Gauge32'},
        'free_pages'                => {mib => 'myFreePages',                   order => 12,  type => 'Gauge32'},
        'database_pages'            => {mib => 'myDatabasePages',               order => 13,  type => 'Gauge32'},
        'modified_pages'            => {mib => 'myModifiedPages',               order => 14,  type => 'Gauge32'},
        'pages_read'                => {mib => 'myPagesRead',                   order => 15,  type => 'Counter32'},
        'pages_created'             => {mib => 'myPagesCreated',                order => 16,  type => 'Counter32'},
        'pages_written'             => {mib => 'myPagesWritten',                order => 17,  type => 'Counter32'},
        'file_fsyncs'               => {mib => 'myFileFsyncs',                  order => 18,  type => 'Counter32'},
        'file_reads'                => {mib => 'myFileReads',                   order => 19,  type => 'Counter32'},
        'file_writes'               => {mib => 'myFileWrites',                  order => 20,  type => 'Counter32'},
        'log_writes'                => {mib => 'myLogWrites',                   order => 21,  type => 'Counter32'},
        'pending_aio_log_ios'       => {mib => 'myPendingAIOLogIOs',            order => 22,  type => 'Gauge32'},
        'pending_aio_sync_ios'      => {mib => 'myPendingAIOSyncIOs',           order => 23,  type => 'Gauge32'},
        'pending_buf_pool_flushes'  => {mib => 'myPendingBufPoolFlushes',       order => 24,  type => 'Gauge32'},
        'pending_chkp_writes'       => {mib => 'myPendingChkpWrites',           order => 25,  type => 'Gauge32'},
        'pending_ibuf_aio_reads'    => {mib => 'myPendingIbufAIOReads',         order => 26,  type => 'Gauge32'},
        'pending_log_flushes'       => {mib => 'myPendingLogFlushes',           order => 27,  type => 'Gauge32'},
        'pending_log_writes'        => {mib => 'myPendingLogWrites',            order => 28,  type => 'Gauge32'},
        'pending_normal_aio_reads'  => {mib => 'myPendingNormalAIOReads',       order => 29,  type => 'Gauge32'},
        'pending_normal_aio_writes' => {mib => 'myPendingNormalAIOWrites',      order => 30,  type => 'Gauge32'},
        'ibuf_inserts'              => {mib => 'myIbufInserts',                 order => 31,  type => 'Counter32'},
        'ibuf_merged'               => {mib => 'myIbufMerged',                  order => 32,  type => 'Counter32'},
        'ibuf_merges'               => {mib => 'myIbufMerges',                  order => 33,  type => 'Counter32'},
        'spin_waits'                => {mib => 'mySpinWaits',                   order => 34,  type => 'Counter32'},
        'spin_rounds'               => {mib => 'mySpinRounds',                  order => 35,  type => 'Counter32'},
        'os_waits'                  => {mib => 'myOsWaits',                     order => 36,  type => 'Counter32'},
        'rows_inserted'             => {mib => 'myRowsInserted',                order => 37,  type => 'Counter32'},
        'rows_updated'              => {mib => 'myRowsUpdated',                 order => 38,  type => 'Counter32'},
        'rows_deleted'              => {mib => 'myRowsDeleted',                 order => 39,  type => 'Counter32'},
        'rows_read'                 => {mib => 'myRowsRead',                    order => 40,  type => 'Counter32'},
        'Table_locks_waited'        => {mib => 'myTableLocksWaited',            order => 41,  type => 'Counter32'},
        'Table_locks_immediate'     => {mib => 'myTableLocksImmediate',         order => 42,  type => 'Counter32'},
        'Slow_queries'              => {mib => 'mySlowQueries',                 order => 43,  type => 'Counter32'},
        'Open_files'                => {mib => 'myOpenFiles',                   order => 44,  type => 'Gauge32'},
        'Open_tables'               => {mib => 'myOpenTables',                  order => 45,  type => 'Gauge32'},
        'Opened_tables'             => {mib => 'myOpenedTables',                order => 46,  type => 'Counter32'},
        'innodb_open_files'         => {mib => 'myInnodbOpenFiles',             order => 47,  type => 'Counter32'},
        'open_files_limit'          => {mib => 'myOpenFilesLimit',              order => 48,  type => 'Counter32'},
        'table_cache'               => {mib => 'myTableCache',                  order => 49,  type => 'Counter32'},
        'Aborted_clients'           => {mib => 'myAbortedClients',              order => 50,  type => 'Counter32'},
        'Aborted_connects'          => {mib => 'myAbortedConnects',             order => 51,  type => 'Counter32'},
        'Max_used_connections'      => {mib => 'myMaxUsedConnections',          order => 52,  type => 'Counter32'},
        'Slow_launch_threads'       => {mib => 'mySlowLaunchThreads',           order => 53,  type => 'Counter32'},
        'Threads_cached'            => {mib => 'myThreadsCached',               order => 54,  type => 'Gauge32'},
        'Threads_connected'         => {mib => 'myThreadsConnected',            order => 55,  type => 'Gauge32'},
        'Threads_created'           => {mib => 'myThreadsCreated',              order => 56,  type => 'Counter32'},
        'Threads_running'           => {mib => 'myThreadsRunning',              order => 57,  type => 'Gauge32'},
        'max_connections'           => {mib => 'myMaxConnections',              order => 58,  type => 'Gauge32'},
        'thread_cache_size'         => {mib => 'myThreadCacheSize',             order => 59,  type => 'Gauge32'},
        'Connections'               => {mib => 'myConnections',                 order => 60,  type => 'Counter32'},
        'slave_running'             => {mib => 'mySlaveRunning',                order => 61,  type => 'Gauge32'},
        'slave_stopped'             => {mib => 'mySlaveStopped',                order => 62,  type => 'Gauge32'},
        'Slave_retried_transactions' => {mib => 'mySlaveRetriedTransactions',   order => 63,  type => 'Counter32'},
        'slave_lag'                  => {mib => 'mySlaveLag',                   order => 64,  type => 'Gauge32'},
        'Slave_open_temp_tables'     => {mib => 'mySlaveOpenTempTables',        order => 65,  type => 'Gauge32'},
        'Qcache_free_blocks'         => {mib => 'myQcacheFreeBlocks',           order => 66,  type => 'Gauge32'},
        'Qcache_free_memory'        => {mib => 'myQcacheFreeMemory',            order => 67,  type => 'Gauge32'},
        'Qcache_hits'               => {mib => 'myQcacheHits',                  order => 68,  type => 'Counter32'},
        'Qcache_inserts'            => {mib => 'myQcacheInserts',               order => 69,  type => 'Counter32'},
        'Qcache_lowmem_prunes'      => {mib => 'myQcacheLowmemPrunes',          order => 70,  type => 'Counter32'},
        'Qcache_not_cached'         => {mib => 'myQcacheNotCached',             order => 71,  type => 'Counter32'},
        'Qcache_queries_in_cache'   => {mib => 'myQcacheQueriesInCache',        order => 72,  type => 'Gauge32'},
        'Qcache_total_blocks'       => {mib => 'myQcacheTotalBlocks',           order => 73,  type => 'Gauge32'},
        'query_cache_size'          => {mib => 'myQueryCacheSize',              order => 74,  type => 'Counter32'},
        'Questions'                 => {mib => 'myQuestions',                   order => 75,  type => 'Counter32'},
        'Com_update'                => {mib => 'myComUpdate',                   order => 76,  type => 'Counter32'},
        'Com_insert'                => {mib => 'myComInsert',                   order => 77,  type => 'Counter32'},
        'Com_select'                => {mib => 'myComSelect',                   order => 78,  type => 'Counter32'},
        'Com_delete'                => {mib => 'myComDelete',                   order => 79,  type => 'Counter32'},
        'Com_replace'               => {mib => 'myComReplace',                  order => 80,  type => 'Counter32'},
        'Com_load'                  => {mib => 'myComLoad',                     order => 81,  type => 'Counter32'},
        'Com_update_multi'          => {mib => 'myComUpdateMulti',              order => 82,  type => 'Counter32'},
        'Com_insert_select'         => {mib => 'myComInsertSelect',             order => 83,  type => 'Counter32'},
        'Com_delete_multi'          => {mib => 'myComDeleteMulti',              order => 84,  type => 'Counter32'},
        'Com_replace_select'        => {mib => 'myComReplaceSelect',            order => 85,  type => 'Counter32'},
        'Select_full_join'          => {mib => 'mySelectFullJoin',              order => 86,  type => 'Counter32'},
        'Select_full_range_join'    => {mib => 'mySelectFullRangeJoin',         order => 87,  type => 'Counter32'},
        'Select_range'              => {mib => 'mySelectRange',                 order => 88,  type => 'Counter32'},
        'Select_range_check'        => {mib => 'mySelectRangeCheck',            order => 89,  type => 'Counter32'},
        'Select_scan'               => {mib => 'mySelectScan',                  order => 90,  type => 'Counter32'},
        'Sort_merge_passes'         => {mib => 'mySortMergePasses',             order => 91,  type => 'Counter32'},
        'Sort_range'                => {mib => 'mySortRange',                   order => 92,  type => 'Counter32'},
        'Sort_rows'                 => {mib => 'mySortRows',                    order => 93,  type => 'Counter32'},
        'Sort_scan'                 => {mib => 'mySortScan',                    order => 94,  type => 'Counter32'},
        'Created_tmp_tables'        => {mib => 'myCreatedTmpTables',            order => 95,  type => 'Counter32'},
        'Created_tmp_disk_tables'   => {mib => 'myCreatedTmpDiskTables',        order => 96,  type => 'Counter32'},
        'Created_tmp_files'         => {mib => 'myCreatedTmpFiles',             order => 97,  type => 'Counter32'},
        'Bytes_sent'                => {mib => 'myBytesSent',                   order => 98,  type => 'Counter64'},
        'Bytes_received'            => {mib => 'myBytesReceived',               order => 99,  type => 'Counter64'},
        'innodb_log_buffer_size'     => {mib => 'myInnodbLogBufferSize',        order => 100, type => 'Gauge32'},
        'unflushed_log'              => {mib => 'myUnflushedLog',               order => 101, type => 'Gauge32'},
        'log_bytes_flushed'          => {mib => 'myLogBytesFlushed',            order => 102, type => 'Counter64'},
        'log_bytes_written'          => {mib => 'myLogBytesWritten',            order => 103, type => 'Counter64'},
        'relay_log_space'            => {mib => 'myRelayLogSpace',              order => 104, type => 'Counter32'},
        'binlog_cache_size'          => {mib => 'myBinlogCacheSize',            order => 105, type => 'Gauge32'},
        'Binlog_cache_disk_use'      => {mib => 'myBinlogCacheDiskUse',         order => 106, type => 'Counter32'},
        'Binlog_cache_use'           => {mib => 'myBinlogCacheUse',             order => 107, type => 'Counter32'},
        'binary_log_space'           => {mib => 'myBinaryLogSpace',             order => 108, type => 'Counter32'},
        'State_closing_tables'       => {mib => 'myStateClosingTables',         order => 109, type => 'Gauge32'},
        'State_copying_to_tmp_table' => {mib => 'myStateCopyingToTmpTable',     order => 110, type => 'Gauge32'},
        'State_end'                 => {mib => 'myStateEnd',                    order => 111, type => 'Gauge32'},
        'State_freeing_items'       => {mib => 'myStateFreeingItems',           order => 112, type => 'Gauge32'},
        'State_init'                => {mib => 'myStateInit',                   order => 113, type => 'Gauge32'},
        'State_locked'              => {mib => 'myStateLocked',                 order => 114, type => 'Gauge32'},
        'State_login'               => {mib => 'myStateLogin',                  order => 115, type => 'Gauge32'},
        'State_preparing'           => {mib => 'myStatePreparing',              order => 116, type => 'Gauge32'},
        'State_reading_from_net'    => {mib => 'myStateReadingFromNet',         order => 117, type => 'Gauge32'},
        'State_sending_data'        => {mib => 'myStateSendingData',            order => 118, type => 'Gauge32'},
        'State_sorting_result'      => {mib => 'myStateSortingResult',          order => 119, type => 'Gauge32'},
        'State_statistics'          => {mib => 'myStateStatistics',             order => 120, type => 'Gauge32'},
        'State_updating'            => {mib => 'myStateUpdating',               order => 121, type => 'Gauge32'},
        'State_writing_to_net'      => {mib => 'myStateWritingToNet',           order => 122, type => 'Gauge32'},
        'State_none'                => {mib => 'myStateNone',                   order => 123, type => 'Gauge32'},
        'State_other'               => {mib => 'myStateOther',                  order => 124, type => 'Gauge32'},
        'additional_pool_alloc'     => {mib => 'myAdditionalPoolAlloc',         order => 125, type => 'Gauge64'},
        'total_mem_alloc'           => {mib => 'myTotalMemAlloc',               order => 126, type => 'Gauge64'},
        'hash_index_cells_total'    => {mib => 'myHashIndexCellsTotal',         order => 127, type => 'Gauge32'},
        'hash_index_cells_used'     => {mib => 'myHashIndexCellsUsed',          order => 128, type => 'Gauge32'},
        'innodb_lock_structs'       => {mib => 'myInnoDBLockStructs',           order => 129, type => 'Gauge32'},
        'innodb_lock_wait_secs'     => {mib => 'myInnoDBLockWaitSecs',          order => 130, type => 'Gauge32'},
        'innodb_tables_in_use'      => {mib => 'myInnoDBTablesInUse',           order => 131, type => 'Gauge32'},
        'innodb_locked_tables'      => {mib => 'myInnoDBLockedTables',          order => 132, type => 'Gauge32'},
        'uncheckpointed_bytes'      => {mib => 'myUncheckpointedBytes',         order => 133, type => 'Gauge32'},
        'ibuf_cell_count'           => {mib => 'myIBufCellCount',               order => 134, type => 'Gauge32'},
        'ibuf_used_cells'           => {mib => 'myIBufUsedCells',               order => 135, type => 'Gauge32'},
        'ibuf_free_cells'           => {mib => 'myIBufFreeCells',               order => 136, type => 'Gauge32'},
        'adaptive_hash_memory'      => {mib => 'myAdaptiveHashMemory',          order => 137, type => 'Gauge64'},
        'page_hash_memory'          => {mib => 'myPageHashMemory',              order => 138, type => 'Gauge64'},
        'dictionary_cache_memory'   => {mib => 'myDictionaryCacheMemory',       order => 139, type => 'Gauge64'},
        'file_system_memory'        => {mib => 'myFileSystemMemory',            order => 140, type => 'Gauge64'},
        'lock_system_memory'        => {mib => 'myLockSystemMemory',            order => 141, type => 'Gauge64'},
        'recovery_system_memory'    => {mib => 'myRecoverySystemMemory',        order => 142, type => 'Gauge64'},
        'thread_hash_memory'        => {mib => 'myThreadHashMemory',            order => 143, type => 'Gauge64'},
        'innodb_sem_waits'          => {mib => 'myInnoDBSemWaits',              order => 144, type => 'Counter64'},
        'innodb_sem_wait_time_ms'   => {mib => 'myInnoDBSemWaitTime',           order => 145, type => 'Counter64'},
        'key_buf_bytes_unflushed'   => {mib => 'myKeyBufBytesUnflushed',        order => 146, type => 'Gauge64'},
        'key_buf_bytes_used'        => {mib => 'myKeyBufBytesUsed',             order => 147, type => 'Gauge64'},
        'key_buffer_size'           => {mib => 'myKeyBufferSize',               order => 148, type => 'Gauge64'},
        'Innodb_row_lock_time'      => {mib => 'myInnoDBRowLockTime',           order => 149, type => 'Counter64'},
        'Innodb_row_lock_waits'     => {mib => 'myInnoDBRowLockWaits',          order => 150, type => 'Counter64'},
    },
    startoid => '.1.3.6.1.4.1.20267.200.1',
}
